test/account.spec.ts

```diff
@@ -16,6 +18,187 @@ import {
 } from '../src'
 const eip1014Testdata = require('./testdata/eip1014Examples.json')
 
+describe('Account', function() {
+  describe('empty constructor', function() {
+    const account = new Account()
+    it('should have zero nonce', function() {
+      assert.ok(account.nonce.isZero())
+    })
+    it('should have zero balance', function() {
+      assert.ok(account.balance.isZero())
+    })
+    it('should have stateRoot equal to KECCAK256_RLP', function() {
+      assert.ok(
+        account.stateRoot.toString('hex'),
+        '56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421',
+      )
+    })
+    it('should have codeHash equal to KECCAK256_NULL', function() {
+      assert.equal(
+        account.codeHash.toString('hex'),
+        'c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470',
+      )
+    })
+  })
+
+  describe('from Array data', function() {
+    const raw = [
+      '0x02', // nonce
+      '0x0384', // balance
+      '0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421', // stateRoot
+      '0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470', // codeHash
+    ]
+    const account = Account.fromValuesArray(raw.map(toBuffer))
+    it('should have correct nonce', function() {
+      assert.ok(account.nonce.eqn(2))
+    })
+    it('should have correct balance', function() {
+      assert.ok(account.balance.eqn(900))
+    })
+    it('should have correct stateRoot', function() {
+      assert.equal(
+        account.stateRoot.toString('hex'),
+        '56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421',
+      )
+    })
+    it('should have correct codeHash', function() {
+      assert.equal(
+        account.codeHash.toString('hex'),
+        'c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470',
+      )
+    })
+  })
+
+  describe('from Object data', function() {
+    const raw = {
+      nonce: '0x02',
+      balance: '0x0384',
+      stateRoot: '0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421',
+      codeHash: '0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470',
+    }
+    const account = Account.fromAccountData(raw)
+    it('should have correct nonce', function() {
+      assert.ok(account.nonce.eqn(2))
+    })
+    it('should have correct balance', function() {
+      assert.ok(account.balance.eqn(900))
+    })
+    it('should have correct stateRoot', function() {
+      assert.equal(
+        account.stateRoot.toString('hex'),
+        '56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421',
+      )
+    })
+    it('should have correct codeHash', function() {
+      assert.equal(
+        account.codeHash.toString('hex'),
+        'c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470',
+      )
+    })
+  })
+
+  describe('from RLP data', function() {
+    const accountRlp = Buffer.from(
```

Comment:
Nit: we could improve this test by explicitly generating the Buffer via `rlp.encode` and then checking against the original data. From this point of view this buffer looks like a magic number 😄

Reply:
yeah we do that in testing `serialize()` a bit further down.

i copied these tests from ethereumjs-account so i'm not sure who generated the original RLP, i guess we can add a comment of the raw data, but i like one example of using a rlp hex string.